src/orb-tools/orb.yml

```diff
@@ -45,53 +185,93 @@ jobs:
     executor: cli/default
     parameters:
+      orb-path:
+        description: "Path to an orb."
+        type: string
       namespace:
-        description: "The namespace where this should be published."
+        description: "Namespace the orb lives under."
         type: string
-      orb:
-        description: "The orb name where this should be published."
+      orb-name:
+        description: "Name of the orb to increment.;"
         type: string
-      label:
-        description: "The label to use when publishing the dev orb (eg: if you pass `foo` you will publish to `dev:foo`)"
+      segment:
+        description: "The semver segment to increment 'major' or 'minor' or 'patch'"
+        type: enum
+        enum: ["major", "minor", "patch"]
+      workspace-path:
+        description: "The directory relative to the workspace root where the orb file will be found. Should end in a trailing /"
+        type: string
+        default: ""
+      token-variable:
+        description: "The env var containing your token. Pass this as a literal string such as `$ORB_PUBLISHING_TOKEN`. Do not paste the actual token into your configuration."
+        type: string
+    steps:
+      - when:
+          condition: << parameters.workspace-path >>
+          steps:
+            - attach_workspace:
+                at: .
+            - increment:
+                orb-path: << parameters.workspace-path >>$(basename << parameters.orb-path >>)
+                namespace: << parameters.namespace >>
+                orb-name: << parameters.orb-name >>
+                segment: << parameters.segment >>
+                token-variable: << parameters.token-variable >>
+      - unless:
+          condition: << parameters.workspace-path >>
+          steps:
+            - checkout
+            - increment:
+                orb-path: << parameters.orb-path >>
+                namespace: << parameters.namespace >>
+                orb-name: << parameters.orb-name >>
+                segment: << parameters.segment >>
+                token-variable: << parameters.token-variable >>
+
+  publish-orb:
+    executor: cli/default
+    parameters:
+      orb-ref:
+        description: >
+          A fully-qualified reference to an orb. This takes the form namespace/orb@version.
+          NOTE: To publish a dev version prefix the version with 'dev:' like so namespace/orb@dev:label
         type: string
       workspace-path:
         description: "The directory relative to the workspace root where the orb file will be found. Should end in a trailing /"
         type: string
         default: ""
       token-variable:
-        description: "The env var containing your token. Pass this as a literal string such as `$ORB_DEV_PUBLISHING_TOKEN`. Do not paste the actual token into your configuration."
+        description: "The env var containing your token. Pass this as a literal string such as `$ORB_PUBLISHING_TOKEN`. Do not paste the actual token into your configuration."
```

Comment:
This was very intentionally different -- we don't anticipate using the same token for both dev and prod publishing.

Reply:
I had originally split the two apart into `publish-release-orb` and `publish-dev-orb` but it felt a bit like hand holding around the `orb-ref`. It's my hope that anyone using this should understand how orb refs work and use that to drive how they want to publish an orb.

Reply:
Yeah, I’m torn. Baking it into the semantics means someone learning orb publishing from this orb is less likely to make a “mistake” as they build a publishing pipeline.